### DISCUSSION TITLE
How many simultaneous a2dp sink/source devices?

### DISCUSSION THREAD

**Discussion by JWZ1996** [Arkq/bluez-alsa#646] [Q&A]
How many audio in/out devices it is possible to connect simultaneously, at the same time via bluealsa?

When I type `bluealsa -p a2dp-sink`, I get:

```
Jun 22 09:02:52 nuc058 bluetoothd[491]: Endpoint registered: sender=:1.77 path=/org/bluez/hci0/A2DP/AAC/sink/1
Jun 22 09:02:52 nuc058 bluetoothd[491]: Endpoint registered: sender=:1.77 path=/org/bluez/hci0/A2DP/AAC/sink/2
Jun 22 09:02:52 nuc058 bluetoothd[491]: Endpoint registered: sender=:1.77 path=/org/bluez/hci0/A2DP/SBC/sink/1
Jun 22 09:02:52 nuc058 bluetoothd[491]: Endpoint registered: sender=:1.77 path=/org/bluez/hci0/A2DP/SBC/sink/2
```

Does it mean that I can have 4 devices, 2 with AAC and 2 with SBC?

**Comment by arkq** [comment]
The number of simultaneously connected devices is not restricted by BlueALSA. However, there might be some limit imposed by Bluetooth/BlueZ. By default BlueALSA registers 2 endpoints per enabled codec. During the connection new endpoint is registered to keep the number of available (not connected) endpoints >= 2 (per codec).

Up to bluez-alsa <= 4.1.x this should work with every BlueZ version. But in the next major/minor release (and soon in the master branch), this functionality will require BlueZ >= 5.67 (which is not released yet).
